src/ResourceManager.cpp

```diff
@@ -94,20 +161,27 @@ ResourceManager* ResourceManager::getInstance()
     return Base::Singleton<ResourceManager>::get();
 }
 
-Format::Dat::Item* ResourceManager::datFileItem(string filename)
+Format::Dat::Item* ResourceManager::datFileItem(const std::string& filename)
 {
-    std::transform(filename.begin(), filename.end(), filename.begin(), ::tolower);
-
     // Return item from cache
     auto itemIt = _datItemMap.find(filename);
     if (itemIt != _datItemMap.end())
     {
         return itemIt->second;
     }
 
+    std::string filenameLowered(filename);
```

Comment:
Um. Why do this twice? It doesn't make sense to lookup non-lowercased filenames in _datItemMap, because we always use lowercased filenames.